src/plugin/plugin.ts

```diff
@@ -843,6 +848,55 @@ class Purchases {
   ) {
     window.cordova.exec(callback, null, PLUGIN_NAME, "checkTrialOrIntroductoryPriceEligibility", [productIdentifiers]);
   }
+  /**
+   * Sets a function to be called on purchases initiated on the Apple App Store. This is only used in iOS.
+   * @param {ShouldPurchasePromoProductListener} shouldPurchasePromoProductListener Called when a user initiates a
+   * promotional in-app purchase from the App Store. If your app is able to handle a purchase at the current time, run
+   * the deferredPurchase function. If the app is not in a state to make a purchase: cache the deferredPurchase, then
+   * call the deferredPurchase when the app is ready to make the promotional purchase.
+   * If the purchase should never be made, you don't need to ever call the deferredPurchase and the app will not
+   * proceed with promotional purchases.
+   */
+  public static addShouldPurchasePromoProductListener(shouldPurchasePromoProductListener: ShouldPurchasePromoProductListener): void { 
+    if (typeof shouldPurchasePromoProductListener !== "function") {
+      throw new Error("addPurchaserInfoUpdateListener needs a function");
+    }
+    shouldPurchasePromoProductListeners.push(shouldPurchasePromoProductListener);
+  }
+
+  /**
+   * Removes a given ShouldPurchasePromoProductListener
+   * @param {ShouldPurchasePromoProductListener} listenerToRemove ShouldPurchasePromoProductListener reference of the listener to remove
+   * @returns {boolean} True if listener was removed, false otherwise
+   */
+  public static removeShouldPurchasePromoProductListener(listenerToRemove: ShouldPurchasePromoProductListener): boolean { 
+    if (shouldPurchasePromoProductListeners.indexOf(listenerToRemove) !== -1) {
+      shouldPurchasePromoProductListeners = shouldPurchasePromoProductListeners.filter(
+        listener => listenerToRemove !== listener
+      );
+      return true;
+    }
+    return false;
+  }
+
+  private static setupShouldPurchasePromoProductCallback() { 
+    window.cordova.exec(
+      (callbackResult: any) => {
+          const callbackID = callbackResult.callbackID;
```

Comment:
This two lines could be removed with:
`({callbackID}:{callbackID: number}) => {`

Reply:
cool! didn't know that! I'll update

Reply:
it'd be `({callbackResult}:{callbackID: number}) = {`, right?